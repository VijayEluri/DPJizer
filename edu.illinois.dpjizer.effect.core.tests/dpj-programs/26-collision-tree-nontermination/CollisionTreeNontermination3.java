/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
class CT<region P1, P2> {
	region L, R;

	protected CT<P1:L, P2> l in P1:L;
	protected CT<P1:R, P2> r in P1:R;

	public CT() {
	}

	public <region Q>void intersect1(CT<Q, Q> ct) {
		this.<region Q, Q>intersect2(ct);
	}
	
	public <region S1, S2>void intersect2(CT<S1, S2> ct) 
	{
		l = null;
		r = null;
		ct.intersect2(l);
		ct.<region P1:R, P2>intersect2(r);
	}
	
}
