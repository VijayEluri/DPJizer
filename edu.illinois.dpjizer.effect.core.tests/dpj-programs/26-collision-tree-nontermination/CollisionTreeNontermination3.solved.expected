CT.<>intersect1(CT<Q,Q>) writes P1 : CT.L : *, P1 : CT.R : *, Q : CT.L : *, Q : CT.R : *
CT.<>intersect2(CT<S1,S2>) writes P1 : CT.L : *, P1 : CT.R : *, S1 : CT.L : *, S1 : CT.R : *
CT.CT() pure
